server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -465,26 +491,56 @@ private void executeSearch(SearchTask task, SearchTimeProvider timeProvider, Sea
                                OriginalIndices localIndices, List<SearchShardIterator> remoteShardIterators,
                                BiFunction<String, String, DiscoveryNode> remoteConnections, ClusterState clusterState,
                                Map<String, AliasFilter> remoteAliasMap, ActionListener<SearchResponse> listener,
-                               SearchResponse.Clusters clusters) {
+                               SearchResponse.Clusters clusters, Map<ShardId, SearchContextIdForNode> searchContexts,
+                               SearchAsyncActionProvider searchAsyncActionProvider) {
 
         clusterState.blocks().globalBlockedRaiseException(ClusterBlockLevel.READ);
+        if (searchRequest.allowPartialSearchResults() == null) {
+            // No user preference defined in search request - apply cluster service default
+            searchRequest.allowPartialSearchResults(searchService.defaultAllowPartialSearchResults());
+        }
+
         // TODO: I think startTime() should become part of ActionRequest and that should be used both for index name
         // date math expressions and $now in scripts. This way all apis will deal with now in the same way instead
         // of just for the _search api
-        final Index[] indices = resolveLocalIndices(localIndices, searchRequest.indicesOptions(), clusterState, timeProvider);
-        Map<String, AliasFilter> aliasFilter = buildPerIndexAliasFilter(searchRequest, clusterState, indices, remoteAliasMap);
-        Map<String, Set<String>> routingMap = indexNameExpressionResolver.resolveSearchRouting(clusterState, searchRequest.routing(),
-            searchRequest.indices());
-        routingMap = routingMap == null ? Collections.emptyMap() : Collections.unmodifiableMap(routingMap);
-        String[] concreteIndices = new String[indices.length];
-        for (int i = 0; i < indices.length; i++) {
-            concreteIndices[i] = indices[i].getName();
-        }
-        Map<String, Long> nodeSearchCounts = searchTransportService.getPendingSearchRequests();
-        GroupShardsIterator<ShardIterator> localShardsIterator = clusterService.operationRouting().searchShards(clusterState,
+        final List<SearchShardIterator> localShardIterators;
+        final Map<String, AliasFilter> aliasFilter;
+        final Map<String, Set<String>> indexRoutings;
+
+        boolean preFilterSearchShards;
+        if (searchContexts != null) {
+            assert searchRequest.searchContextBuilder() != null;
+            aliasFilter = new HashMap<>();
+            for (ShardId shardId : searchContexts.keySet()) {
+                aliasFilter.put(shardId.getIndex().getUUID(), AliasFilter.EMPTY);
```

Comment:
Do we expect a search request with `search_context` work with filtered alias?  If not, may be we should explain in the `search_context.asciidoc` that users should not open search contexts on filtered aliases.
Because currently  we can use filtered alias in an open search context request, but if we use this search context in search requests, the search results will not be filtered.

Reply:
Good catch, this is not implemented. We should add the filtered aliases in the reader context, I'll try to tackle this next week.

Reply:
Implemented in https://github.com/elastic/elasticsearch/pull/56480/commits/96fe51f860d5aa5184aad53bbf7101b8d5671b0f